src/main/java/blusunrize/immersiveengineering/common/data/Recipes.java

```diff
@@ -1488,7 +1492,7 @@ private void add3x3Conversion(IItemProvider bigItem, Tag<?> bigTag, IItemProvide
 			.addCriterion("has_"+toPath(smallItem), hasItem(smallItem))
 			.build(out, toRL(toPath(smallItem)+"_to_")+toPath(bigItem));
 		ShapelessRecipeBuilder.shapelessRecipe(smallItem, 9)
-			.addIngredient((Tag<Item>) bigTag)
+			.addIngredient(makeIngredient(bigTag))
```

Comment:
I'm still not happy with this. There's still a cast between block and item tags going on, it's just hidden a few layers away. You should really use `IETags#getItemTag`. And `makeIngredient` will need a rework as well (the `Object` parameter needs to go, and the cast to raw `Tag`s as well), but that's not something for this PR.

Reply:
Problem with `IETags#getItemTag` and `Tag<?> bigTag` is that the `bigTag` is being used for both `Tag<Item>` and `Tag<Block>` but `IETags#getItemTag` needs `Tag<Block>` so `Tag<?>` won't work then either.

When i change `Tag<?> bigTag` to one or the other, the precompiler cries around in 1 or 2 places.

So it's either i have to either leave what i had befor or use `makeIngredient`

Reply:
Tested `IETags#getItemTag((Tag<Block>) bigTag)` and it just results in a crash during generation.

Reply:
Doing either `.addIngredient((Tag<Item>) bigTag)` or `.addIngredient(makeIngredient(bigTag))` without throwing up.

Reply:
No matter what i do, there _will_ be a cast somewhere. Otherwise it just wont work.

Reply:
Nevermind im dumb. Fixed.